### PR TITLE
Hide ignored lines in rustdoc

### LINKED ISSUE #1620: Rustdoc ignored lines are shown in completion and hover
image
image

### PR DESCRIPTION
**Pull request by rizakrko**
This fixes #1620.

### CODE REVIEW

File: crates/ra_lsp_server/src/markdown.rs [hunk 1]
```diff
@@ -1,7 +1,11 @@
-pub(crate) fn mark_fenced_blocks_as_rust(src: &str) -> String {
+pub(crate) fn format_docs(src: &str) -> String {
     let mut processed_lines = Vec::new();
     let mut in_code_block = false;
     for line in src.lines() {
+        if in_code_block && line.trim_start().starts_with("# ") {
```
Comment (lnicola) [inline crates/ra_lsp_server/src/markdown.rs]: Hmm, this is interesting. It looks like lines need to start with `# ` (the space is required), but lines with only a `#` are also hidden.

Comment (rizakrko) [reply crates/ra_lsp_server/src/markdown.rs]: Hm, I am not sure about this.
If rust section of docs contains `#[derive(Foo)]`, it is not ignored by rustdoc (I might be wrong here)

Comment (lnicola) [reply crates/ra_lsp_server/src/markdown.rs]: I meant this case:

```rust
/// ```
/// # use std::collections::HashMap;
/// #
/// // the line above doesn't show up in rustdoc, even if it doesn't start with "# "
/// fn foo() { }
/// ```
```

Comment (lnicola) [reply crates/ra_lsp_server/src/markdown.rs]:
```rust
if in_code_block {
    let trimmed = line.trim();
    if trimmed.starts_with("# ") || trimmed == "#" {
        continue;
    }
```

Comment (rizakrko) [reply crates/ra_lsp_server/src/markdown.rs]: Oh, of course you're right, i missed this case.

Comment (lnicola) [reply crates/ra_lsp_server/src/markdown.rs]: There's also the case of `/// #     `, of course (spaces at the end).

Comment (rizakrko) [reply crates/ra_lsp_server/src/markdown.rs]: I'v updated PR. The only this I am not sure about is tabs. Can they appear at this stage?

Comment (lnicola) [reply crates/ra_lsp_server/src/markdown.rs]: No idea. The interesting case is probably `"/// #\tx"`.

Comment (rizakrko) [reply crates/ra_lsp_server/src/markdown.rs]: Fixed

**Comment by kjeremy** [comment]
This LGTM.

As for your question on `\t` this is how rls does it: https://github.com/rust-lang/rls/blob/9bbe734d730bbd30dbd3ab6adba1a09aa5fb4187/rls/src/actions/hover.rs#L30 which is more complete (and complicated) but we can tackle those as they come up.

**Comment by lnicola** [comment]
Can't we lift that code from RLS?

**Comment by rizakrko** [comment]
I'm not sure if all logic is required, but I'll try to reuse some parts of it.

**Comment by kjeremy** [comment]
I wouldn't worry about implementing all of it now.

**Pull request merged**